From ea7ff3a90d902cc87ac11bf2ea9db1945c443da1 Mon Sep 17 00:00:00 2001
From: Sathesh Edara <sedara@marvell.com>
Date: Fri, 17 Mar 2023 13:16:19 +0530
Subject: [PATCH 328/955] net/octeon_ep: set secondary process dev ops

attach eth dev ops to secondary process in order
to fetch device info via dpdk-proc-info application.

Signed-off-by: Sathesh Edara <sedara@marvell.com>
Change-Id: I09d33eb9575045e60195625ec353bba9b1cfb212
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/99374
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Veerasenareddy Burru <vburru@marvell.com>
---
 drivers/net/octeon_ep/otx_ep_ethdev.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/net/octeon_ep/otx_ep_ethdev.c b/drivers/net/octeon_ep/otx_ep_ethdev.c
index 4e1715ee82302..abca7014ac6eb 100644
--- a/drivers/net/octeon_ep/otx_ep_ethdev.c
+++ b/drivers/net/octeon_ep/otx_ep_ethdev.c
@@ -209,6 +209,11 @@ otx_ep_dev_close(struct rte_eth_dev *eth_dev)
 	struct otx_ep_device *otx_epvf;
 	uint32_t num_queues, q;
 
+	if (rte_eal_process_type() != RTE_PROC_PRIMARY) {
+		eth_dev->dev_ops = NULL;
+		return 0;
+	}
+
 	otx_epvf = OTX_EP_DEV(eth_dev);
 	otx_ep_mbox_send_dev_exit(eth_dev);
 	otx_ep_mbox_disable_interrupt(otx_epvf);
@@ -614,8 +619,10 @@ static const struct eth_dev_ops otx_ep_eth_dev_ops = {
 static int
 otx_ep_eth_dev_uninit(struct rte_eth_dev *eth_dev)
 {
-	if (rte_eal_process_type() != RTE_PROC_PRIMARY)
+	if (rte_eal_process_type() != RTE_PROC_PRIMARY) {
+		eth_dev->dev_ops = NULL;
 		return 0;
+	}
 
 	eth_dev->dev_ops = NULL;
 	eth_dev->rx_pkt_burst = NULL;
@@ -664,8 +671,10 @@ otx_ep_eth_dev_init(struct rte_eth_dev *eth_dev)
 	struct rte_ether_addr vf_mac_addr;
 
 	/* Single process support */
-	if (rte_eal_process_type() != RTE_PROC_PRIMARY)
+	if (rte_eal_process_type() != RTE_PROC_PRIMARY) {
+		eth_dev->dev_ops = &otx_ep_eth_dev_ops;
 		return 0;
+	}
 
 	otx_epvf->sdp_packet_mode = SDP_PACKET_MODE_LOOP;
 	rte_eth_copy_pci_info(eth_dev, pdev);
-- 
2.25.1

